c7n/resources/efs.py

```diff
@@ -39,7 +41,34 @@ class resource_type(TypeInfo):
         filter_type = 'scalar'
         universal_taggable = True
 
-    augment = universal_augment
+    def get_source(self, source_type):
+        return DescribeElasticFileSystem(self)
+
+
+class DescribeElasticFileSystem(DescribeSource):
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('efs')
```

Comment:
so we don't generally do this, re augments doing anything besides tags and innate attributes, as it results in extraneous api calls even if the policy doesn't care about the attribute being augmented. instead we do a dedicated filter on the additional api call.

Reply:
Just to make sure I understand...

So the filter should use the local session to send the request only when it is used, instead of always calling the policy, right?

Reply:
yeah.. if we do retrieval in the filter, the apis will only be invoked if the filter is in the policy. there's also an option of the filter caching the fetched data on the resource in an annotation, but first step would just be restructuring/removing the augment and doing the api call in the filter.